js/interface.js

```diff
@@ -909,9 +1002,65 @@ $('#app')
   .on('click', '.find-icon', function() {
     $('.filter-form .form-control').trigger('focus');
   })
-  .on('show.bs.tab', function (e) {
-    var confirmData;
+  .on('click', '[data-back-to-versions]', function (e) {
+    e.preventDefault();
+
+    $versionContents.addClass('hidden').html('');
+    $('#versions-details').removeClass('hidden');
+  })
+  .on('click', '[data-version-preview]', function (e) {
+    e.preventDefault();
+    var id = $(this).data('version-preview');
+    var version = _.find(currentDataSourceVersions, { id: id });
+
+    previewVersion(version);
+  })
+  .on('click', '[data-version-restore]', function (e) {
+    e.preventDefault();
+    var id = $(this).data('version-restore');
+
+    return Fliplet.Modal.confirm({
+      message: 'Are you sure you want to restore this version on your Data Source? This will replace its entire contents.'
+    }).then(function (result) {
+      if (!result) {
+        return;
+      }
+
+      $('#versions').addClass('hidden');
+      $('#versions-details').removeClass('hidden');
+
+      return Fliplet.API.request({
+        url: 'v1/data-sources/' + currentDataSourceId + '/versions/' + id + '/restore',
+        method: 'POST'
+      }).then(function () {
+        return fetchCurrentDataSourceEntries();
+      }).then(function () {
+        $('#show-entries').click();
 
+        Fliplet.Modal.alert({
+          title: 'Version restored',
+          message: 'The version has been restored to your Data Source.'
+        });
+      });
+    });
+  })
+  .on('click', '[data-version-copy]', function (e) {
+    e.preventDefault();
+    var id = $(this).data('version-copy');
+
+    return createDataSource({
+      version: {
+        dataSourceId: currentDataSourceId,
+        id: id
+      }
+    }).then(function () {
```

Comment:
Can we pass to this function `Copy of $dataSourceName` as the default name? Users can then change it in the prompt.

Reply:
done